Documentation/templates/json/swift3.md

```diff
@@ -0,0 +1,52 @@
+## Template Information
+
+| Name      | Description       |
+| --------- | ----------------- |
+| File name | json/swift4.stencil |
+| Invocation example | `swiftgen json -t swift4 …` |
+| Language | Swift 4 |
+| Author | David Jennes |
+
+## When to use it
+
+- When you need to generate *Swift 4* code
+
+## Customization
+
+You can customize some elements of this template by overriding the following parameters when invoking `swiftgen` in the command line, using `--param <paramName>=<newValue>`
+
+| Parameter Name | Default Value | Description |
+| -------------- | ------------- | ----------- |
+| `documentName` | `Document` | Allows you to change the prefix of the generated `enum` for each document. |
+| `enumName` | `Plist` | Allows you to change the name of the generated `enum` containing all files. |
+| `publicAccess` | N/A | If set, the generated constants will be marked as `public`. Otherwise, they'll be declared `internal`. |
+
+## Generated Code
+
+**Extract:**
+
+```swift
+internal enum JSON {
```

Comment:
The `JSON` namespace could be renamed to default to `JSONFiles` instead. Both because it better describes this (`JSONFiles.Sequence.items` feels more clear than `JSON.Sequence.items` to me), but also to avoid high risk of conflicts with an existing `JSON` type in user's code, as it's quite a generic and common term.

Same goes for Plist and YAML of course.